File: demos_code_J2_conception_objet/06_break_package_cycle/avant/src/com/acme/orders/OrderService.java
package com.acme.orders;
import com.acme.billing.BillingService;

public class OrderService {
    private final BillingService billing;
    public OrderService(BillingService billing) { this.billing = billing; }
    public void cancel(String id) {
        System.out.println("orders: cancel " + id);
        billing.refund(id);
    }
    public void markRefunded(String id) { System.out.println("orders: refunded " + id); }
}

File: Demo en Python/demos_code_J2_conception_objet_python/06_break_package_cycle/avant/src/com/acme/orders/order_service.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from com.acme.billing.billing_service import BillingService


class OrderService:
    # Dépendance orders -> billing.
    def __init__(self, billing: "BillingService") -> None:
        self._billing = billing

    def cancel(self, order_id: str) -> None:
        print(f"orders: cancel {order_id}")
        self._billing.refund(order_id)

    def mark_refunded(self, order_id: str) -> None:
        print(f"orders: refunded {order_id}")
